### DISCUSSION TITLE
persistent issues uploading ARW files (the RAW format used by Sony cameras)

### DISCUSSION THREAD

**Discussion by samirugamage** [ente-io/ente#653] [Enhancements]
Failure to Generate Thumbnails: The application often fails to generate thumbnails for the ARW files,with an error message "Thumbnail generation failed"
Upload Failures: Occasionally, the upload process itself fails, with an error message "failed to upload"
This issue has been consistent on a Windows 11 platform.

**Comment by vishnukvmd** [comment]
The best way forward is to integrate [ImageMagick](https://github.com/ImageMagick/ImageMagick) on Windows (available right now on Linux and Mac builds).

IIRC building from source is where we got stuck the last time around, it might be time to revisit this.

Will also help with ente-io/ente#624.

**Comment by samirugamage** [comment]
**### Batch Script for Converting ARW to JPEG**

**Prerequisites**
ImageMagick: Ensure [ImageMagick](https://imagemagick.org/) is installed on your system. ImageMagick is used for the image conversion process.

ExifTool: Download [ExifTool](https://exiftool.org/) by Phil Harvey. You'll need a portable version (exiftool(-k).exe). Rename the downloaded executable to exiftool.exe for convenience.

Setup Steps
Place ExifTool: Move the exiftool.exe file to a known location on your system. For this guide, we'll assume you've placed it in C:\Users\samiru\Downloads\exiftool-12.80\.

Install ImageMagick: If not already installed, download and install ImageMagick from here. During installation, ensure that the option to add ImageMagick to the system's PATH is selected.

Create Batch File: Open Notepad or your preferred text editor, and paste the following script:


```
@echo off
SETLOCAL

:: Set the path to your exiftool.exe here
SET "exifToolPath=C:\Users\samiru\Downloads\exiftool-12.80\exiftool.exe"

echo Processing ARW files in the current directory...

:: Convert ARW files to JPEG with 100% quality
FOR %%G IN (*.ARW) DO (
    echo Converting "%%G"...
    magick "%%G" -quality 100 "%%~nG.jpg"
    echo Embedding metadata from "%%G" to "%%~nG.jpg"...
    "%exifToolPath%" -TagsFromFile "%%G" -all:all -overwrite_original "%%~nG.jpg"
)

echo Conversion complete.

ENDLOCAL
pause
```
Save the Batch File: Save the file with a .bat extension, for example, ConvertARWtoJPEG.bat.

### **Usage**


1.Copy the Batch File: Place the ConvertARWtoJPEG.bat file in the folder containing your ARW files.

2.Run the Batch File: Double-click the ConvertARWtoJPEG.bat file. A Command Prompt window will open, and the script will start converting the ARW files to JPEG format with 100% quality and copying all metadata.

3. Completion: Wait for the script to process all files. The Command Prompt will display a "Conversion complete." message. Press any key to close the Command Prompt.


Notes
The script processes all .ARW files in the directory where the batch file is located.
Ensure the exifToolPath variable in the script matches the actual path where exiftool.exe is located on your system.
